File: Chavda_23B0368/Week_5/Roman_to_integer.java
package Week_5;
import java.util.HashMap;

class Solution {
    public int romanToInt(String s) {
        HashMap<Character, Integer> map = new HashMap<>();
        map.put('I', 1);
        map.put('V', 5);
        map.put('X', 10);
        map.put('L', 50);
        map.put('C', 100);
        map.put('D', 500);
        map.put('M', 1000);

        int sum = 0;
        for(int i = 0; i < s.length(); i++) {
            if((i+1 < s.length()) && (map.get(s.charAt(i)) < map.get(s.charAt(i+1))))
                sum -= map.get(s.charAt(i));
            else
                sum += map.get(s.charAt(i));
        }

        return sum;
    }
}

File: Shreshtha_24b1033/week5/roman_to_integer.cpp
class Solution {
public:
    int romanToInt(string s) {
        unordered_map<char,int> rti={
            {'I', 1},
            {'V', 5},
            {'X', 10},
            {'L', 50},
            {'C', 100},
            {'D', 500},
            {'M', 1000}
        };
        int n=s.length();
        int num=0;
        for(int i=0;i<n-1;i++){
            if(rti[s[i]]>=rti[s[i+1]])num+=rti[s[i]];
            else num-=rti[s[i]];
        }
        num+=rti[s[n-1]];
        return num;
    }
};